controller/controller.go

```diff
@@ -263,30 +306,28 @@ func splitProviderID(id string) (string, string) {
 // assumes the default provider block that is empty.
 //
 // terraform_provider "name" { }
-func getProvider(providers *config.TerraformProviderConfigs, id string) map[string]interface{} {
+func getProvider(providers []hcltmpl.NamedBlock, id string) hcltmpl.NamedBlock {
 	name, alias := splitProviderID(id)
 
-	for _, p := range *providers {
+	for _, p := range providers {
 		// Find the provider by name
-		if pRaw, ok := (*p)[name]; ok {
-			if alias == "" {
-				return *p
-			}
+		if p.Name != name {
+			continue
+		}
 
-			// Find the provider by alias
-			pConf, ok := pRaw.(map[string]interface{})
-			if !ok {
-				// Not much we can do if the provider block has unexpected structure
-				// at this point. We'll move forward with the default empty block.
-				break
-			}
+		if alias == "" {
+			return p
+		}
 
-			a, ok := pConf["alias"].(string)
-			if ok && a == alias {
-				return *p
-			}
+		// Match by alias
+		a, ok := p.Variables["alias"]
```

Comment:
The "a" variable, I'm sure, is for "alias"... but I can't not read this without hearing it as the phrase "A-ok". :)

Reply:
haha. not my proudest variable name but the function is short and seemed a-ok :D